packages/components/managed-data-table/src/data-table-settings/column-manager/column-manager.js

```diff
@@ -0,0 +1,252 @@
+import AsyncSelectInput from '@commercetools-uikit/async-select-input';
+import CollapsibleMotion from '@commercetools-uikit/collapsible-motion';
+import FieldLabel from '@commercetools-uikit/field-label';
+import Spacings from '@commercetools-uikit/spacings';
+import Text from '@commercetools-uikit/text';
+import {
+  CloseIcon,
+  EyeCrossedIcon,
+  EyeIcon,
+  SearchIcon,
+} from '@commercetools-uikit/icons';
+import AccessibleButton from '@commercetools-uikit/accessible-button';
+import { filterDataAttributes } from '@commercetools-uikit/utils';
+import DroppablePanel from '../droppable-panel';
+import {
+  Container,
+  HeaderContainer,
+  DroppableContainer,
+} from '../data-table-settings-panel.styles';
+import messages from './messages';
+import { HIDDEN_COLUMNS_PANEL, SELECTED_COLUMNS_PANEL } from './constants';
+
+export const handleDragEnd = (
+  dragResult,
+  onUpdateColumns,
+  selectedColumns,
+  availableColumns,
+  setIsDragging
+) => {
+  setIsDragging(false);
+  // Invalid drop destination, do nothing
+  if (!dragResult.destination) return;
+  if (dragResult.destination.droppableId === HIDDEN_COLUMNS_PANEL) {
+    if (dragResult.source.droppableId === HIDDEN_COLUMNS_PANEL) return;
+    onUpdateColumns([
+      ...selectedColumns.slice(0, dragResult.source.index),
+      ...selectedColumns.slice(dragResult.source.index + 1),
+    ]);
+  } else {
+    // the destination is the selected columns panel
+
+    // it's a swap when the source and the destination are the same
+    const isSwap = dragResult.source.droppableId === SELECTED_COLUMNS_PANEL;
+
+    const items = isSwap
+      ? // remove the dragged item from its position if it is not coming from
+        // the hidden section (it is a swap)
+        [
+          ...selectedColumns.slice(0, dragResult.source.index),
+          ...selectedColumns.slice(dragResult.source.index + 1),
+        ]
+      : selectedColumns;
+
+    const columns = isSwap ? selectedColumns : availableColumns;
+    const draggedColumn = columns.find(
+      (col) => col.key === dragResult.draggableId
+    );
+
+    // push the column in the new position
+    onUpdateColumns([
+      ...items.slice(0, dragResult.destination.index),
+      draggedColumn,
+      ...items.slice(dragResult.destination.index),
+    ]);
+  }
+};
+
+export const ColumnManager = (props) => {
+  const intl = useIntl();
+  const [isDragging, setIsDragging] = useState(false);
+
+  const handleDragStart = () => {
+    setIsDragging(true);
+  };
+
+  const hiddenColumns = differenceWith(
+    props.availableColumns,
+    props.selectedColumns,
+    (a, b) => a.key === b.key
+  );
+
+  const selectInputComponents = {
+    Option: () => null,
+    Menu: () => null,
+    // eslint-disable-next-line react/display-name
+    DropdownIndicator: () => (
+      <Spacings.Inline alignItems="center">
+        <SearchIcon scale="medium" color="primary" />
+      </Spacings.Inline>
+    ),
+  };
```

Comment:
This should be defined outside the component.
Also it would be better to have static references to components, instead of inline.

```js
const DropdownIndicator = () => (
  <Spacings.Inline alignItems="center">
    <SearchIcon scale="medium" color="primary" />
  </Spacings.Inline>
);
const Nothing = () => null;
const selectInputComponents = {
  Option: Nothing,
  Menu: Nothing,
  DropdownIndicator,
};

```